src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -1398,49 +1412,61 @@ private void StartLoggingEventProcessing()
             void LoggingEventProc()
             {
                 var completeAdding = _loggingEventProcessingCancellation.Token;
-                WaitHandle[] waitHandlesForNextEvent = { completeAdding.WaitHandle, _enqueueEvent };
+                WaitHandle[] waitHandlesForNextEvent = [completeAdding.WaitHandle, _enqueueEvent];
 
-                do
+                try
                 {
-                    if (_eventQueue.TryDequeue(out object ev))
+                    do
                     {
-                        LoggingEventProcessor(ev);
-                        _dequeueEvent.Set();
-                    }
-                    else
-                    {
-                        _emptyQueueEvent.Set();
+                        // Check if instance fields are nulled (cleanup was called)
+                        if (_eventQueue == null || _dequeueEvent == null || _emptyQueueEvent == null || _enqueueEvent == null)
```

Comment:
These can still get  nullified right after checking.
We should store them in local variables and check and use those here. Then nullyfying in the close method could be kept

Reply:
let me know if it looks better now

Reply:
@rainersigwald please join the discussion